src/sentry/integrations/github/client.py

```diff
@@ -264,13 +282,20 @@ def _populate_trees(self, repositories: List[Dict[str, str]]) -> Dict[str, RepoT
                     repo_info, only_use_cache, (3600 * 24) + (3600 * (index % 24))
                 )
             except ApiError as error:
-                self._populate_trees_process_error(error, extra)
+                should_count_error = self._populate_trees_process_error(error, extra)
+                if should_count_error:
+                    connection_error_count += 1
             except Exception:
                 # Report for investigation but do not stop processing
                 logger.exception(
                     "Failed to populate_tree. Investigate. Contining execution.", extra=extra
                 )
```

Comment:
I wonder if we should also increase the tally here. What do you think?

It does not seem to happen too often:
https://sentry.sentry.io/issues/?project=1&query=Failed+to+populate_tree&referrer=issue-list&sort=freq&statsPeriod=90d

Reply:
The type of error request raises is different:
<img width="340" alt="image" src="https://user-images.githubusercontent.com/44410/225114522-e828a598-a010-4e1a-a1ed-d985f8dd13c0.png">
vs
<img width="418" alt="image" src="https://user-images.githubusercontent.com/44410/225114882-73cfe2bf-82c6-4ad5-8865-b45920b0b45c.png">

Perhaps we should catch that error and wrap it in an ApiError:
<img width="403" alt="image" src="https://user-images.githubusercontent.com/44410/225114693-32974e46-d125-4ce9-bedb-c3f1243991ea.png">

We can address it on this PR or a follow up. Whatever you prefer.

Reply:
> Perhaps we should catch that error and wrap it in an ApiError:

Yeah, that's really what I meant - I was surprised that any error thrown during an outgoing request wouldn't be wrapped in an `ApiError`. Anyway, yeah, let's do it.